src/Jupyter/Extensions.cs

```diff
@@ -174,5 +175,17 @@ public static string Dedent(this string text)
             var leftTrimRegex = new Regex(@$"^[ \t]{{{minWhitespace}}}", RegexOptions.Multiline);
             return leftTrimRegex.Replace(text, "");
         }
+
+        /// <summary>
+        ///      Retrieves and JSON-decodes the value for the given parameter name.
+        /// </summary>
+        public static T DecodeParameter<T>(this Dictionary<string, string> parameters, string parameterName)
+        {
+            if (!parameters.TryGetValue(parameterName, out string parameterValue))
+            {
+                return default(T);
```

Comment:
As per @efratshabtai's comment above, maybe an optional parameter here specifying a default other than `default(T)` would be helpful?

Reply:
Agreed, I'll add this.

---
In reply to: [427476716](https://github.com/microsoft/iqsharp/pull/144#discussion_r427476716) [](ancestors = 427476716)